runtime/domain.c

```diff
@@ -401,67 +401,68 @@ asize_t caml_norm_minor_heap_size (intnat wsize)
 
 /* The current minor heap layout is as follows:
 
-- A contiguous memory block of size
-   [caml_minor_heap_max_wsz * caml_params->max_domains]
-  is reserved by [caml_init_domains]. The boundaries
-  of this reserved area are stored in the globals
+- The 'minor heaps reservation' is a contiguous address space of size
+    [caml_minor_heap_max_wsz * caml_params->max_domains]
+  reserved by [caml_init_domains]. Its boundaries are
     [caml_minor_heaps_start]
   and
     [caml_minor_heaps_end].
 
-- Each domain gets a reserved section of this block
-  of size [caml_minor_heap_max_wsz], whose boundaries are stored as
-    [domain_self->minor_heap_area_start]
+- Each domain gets a 'minor heap reservation', a segment of the global
+  reservation of size [caml_minor_heap_max_wsz], whose boundaries are
+    [domain_self->minor_heap_reservation_start]
   and
-    [domain_self->minor_heap_area_end]
+    [domain_self->minor_heap_reservation_end]
 
-  These variables accessed in [stw_resize_minor_heap_reservation],
+  These variables are accessed in [stw_resize_minor_heaps_reservation],
   synchronized by a global barrier.
 
-- Each STW-participating domain then commits a segment of size
-    [domain_state->minor_heap_wsz]
-  starting at
-    [domain_state->minor_heap_area_start]
-  that it actually uses.
-
-  This is done below in
-    [allocate_minor_heap]
-  which is called both at domain-initialization (by [domain_create])
-  and if a request comes to change the minor heap size.
-
-  The boundaries of this committed memory segment are
-     [domain_state->young_start]
+- Each STW-participating domain then commits their 'minor heap' within
+  their minor heap reservation. The minor heap has size
+  [domain_state->minor_heap_wsz], and its boundaries are
+    [domain_state->young_start]
    and
      [domain_state->young_end].
 
   Those [young_{start,end}] variables are never accessed by another
   domain, so they need no synchronization.
 
-- Domains decommit their minor heap segment before leaving the set of
-  STW participants. Otherwise, their minor heap segments could get
-  desynchronized with future changes in the reserved minor heap
-  area. This is done by calling [free_minor_heap].
+  Domains commit their minor heap in
+    [allocate_minor_heap]
+  which is called both at domain-initialization (by [domain_create])
+  and if a request comes to change the minor heap size.
+
+  Domains decommit their minor heap before leaving the set of STW
+  participants. Otherwise, their minor heap could get desynchronized
```

Comment:
The sentence starting with "Otherwise, " seem difficult to appreciate here. In a sense, that's describing a behaviour that was buggy in the past, which has been fixed now. I feel that it is not necessary here.